src/Fixer/Whitespace/MethodChainingIndentationFixer.php

```diff
@@ -0,0 +1,175 @@
+                    $tokens[$index - 1] = new Token([T_WHITESPACE, $tokens[$index - 1]->getContent().$lineEnding]);
+                    --$index;
+                    continue;
+                }
+
+                $prevIndex = $index - 1;
+                $prevToken = $tokens[$prevIndex];
+                $currentWhitespaces = $this->getCurrentWhitespaces($prevToken->getContent());
+
+                if (null !== $currentWhitespaces) {
+                    $prevMeaningIndex = $tokens->getPrevMeaningfulToken($index);
+                    $rightWhitespaces = $this->getRightIndents($prevMeaningIndex, $tokens);
+
+                    if ($currentWhitespaces !== $rightWhitespaces) {
+                        $tokens[$prevIndex] = new Token([T_WHITESPACE, $lineEnding.$rightWhitespaces]);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return string
+     */
+    private function getRightIndents($index, Tokens $tokens)
+    {
+        $indent = $this->whitespacesConfig->getIndent();
+
+        for ($i = $index; $i >= 0; --$i) {
+            if ($i > 0) {
+                $codeToFindIndents = $tokens->generatePartialCode($i - 1, $i);
+            } else {
+                $codeToFindIndents = $tokens[$i]->getContent();
+            }
+
+            $currentWhitespaces = $this->getCurrentWhitespaces($codeToFindIndents);
+
+            if (null !== $currentWhitespaces) {
+                if ($tokens[$i + 1]->isGivenKind(T_OBJECT_OPERATOR) || $this->isMultiLineMethod($i, $index, $tokens)) {
+                    return $currentWhitespaces;
+                }
+
+                return $currentWhitespaces.$indent;
+            }
+        }
+
+        return $indent;
+    }
+
+    /**
+     * @param int    $index
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function needLineBreak($index, Tokens $tokens)
+    {
+        $prevMeaningful = $tokens->getPrevMeaningfulToken($index);
+        $isComment = false;
+
+        for ($i = $index; $i > $prevMeaningful; --$i) {
+            if ($tokens[$i]->isGivenKind(T_OBJECT_OPERATOR) || null !== $this->getCurrentWhitespaces($tokens[$i]->getContent())) {
+                return $isComment;
+            }
+
+            if ($tokens[$i]->isComment()) {
+                $isComment = true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $content
+     *
+     * @return string|null
+     */
+    private function getCurrentWhitespaces($content)
+    {
+        if (1 === preg_match('/\R{1}([ \t]*)$/', $content, $matches)) {
+            return $matches[1];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param int    $start
+     * @param int    $end
+     * @param Tokens $tokens
+     *
+     * @return bool
+     */
+    private function isMultiLineMethod($start, $end, Tokens $tokens)
```

Comment:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2617#discussion_r122853242

it is not checking is it a method at all.
it's checking is there some parenthesis matching (even outside of `start-end` scope!), which doesn't mean it's a method. what about `array(       )` ?

it's not checking is it multiline at all